inc/Engine/Optimization/RUCSS/APIClient.php

```diff
@@ -0,0 +1,85 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS;
+
+class APIClient {
+	/**
+	 * API URL.
+	 */
+	const API_URL = 'https://central-saas.wp-rocket.me:30443/';
+
+	/**
+	 * SAAS main API path.
+	 */
+	const TREESHAKE_PATH = 'api';
+
+	/**
+	 * API Warmup path.
+	 */
+	const WARMUP_PATH = 'warmup';
+
+	/**
+	 * Calls Central Saas API.
+	 *
+	 * @param string $html    HTML content.
+	 * @param string $url     HTML url.
+	 * @param array  $options Array with options sent to Saas API.
+	 *
+	 * @return array
+	 */
+	public function optimize( string $html, string $url, array $options ) : array {
+		$args = [
+			'body'    => [
+				'html'   => $html,
+				'url'    => $url,
+				'config' => $options,
+			],
+			'timeout' => 30,
+		];
+
+		$request = wp_remote_post(
+			self::API_URL . self::TREESHAKE_PATH,
+			$args
+		);
+
+		$error_request = $this->handle_request_error( $request );
+		if ( ! empty( $error_request ) ) {
+			return $error_request;
+		}
+
+		$result = json_decode( wp_remote_retrieve_body( $request ), true );
```

Comment:
We should have a check on the return value of `json_decode()`, in case it fails and returns something that is not an array.